api-consumption/src/main/java/org/zalando/nakadi/service/ConsumptionKpiCollector.java

```diff
@@ -0,0 +1,101 @@
+package org.zalando.nakadi.service;
+
+import org.json.JSONObject;
+import org.zalando.nakadi.security.Client;
+import org.zalando.nakadi.service.publishing.NakadiKpiPublisher;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class ConsumptionKpiCollector {
+    private final String clientId;
+    private final String clientRealm;
+    private final String appNameHashed;
+    private final NakadiKpiPublisher kpiPublisher;
+    private final String kpiEventType;
+    private final long kpiFlushInterval;
+
+    private final Map<String, StreamKpiData> kpiDataPerEventType = new HashMap<>();
+    private long lastKpiEventSent = System.currentTimeMillis();
+
+    public ConsumptionKpiCollector(
+            final Client client,
+            final NakadiKpiPublisher kpiPublisher,
+            final String kpiEventType,
+            final long kpiFlushIntervalMs) {
+        this.clientId = client.getClientId();
+        this.clientRealm = client.getRealm();
+        this.appNameHashed = kpiPublisher.hash(clientId);
+        this.kpiPublisher = kpiPublisher;
+        this.kpiEventType = kpiEventType;
+        this.kpiFlushInterval = kpiFlushIntervalMs;
+    }
+
+    public void sendKpi() {
+        kpiDataPerEventType.forEach(this::publishKpi);
+        kpiDataPerEventType.clear();
+    }
+
+    public void checkAndSendKpi() {
+        if ((System.currentTimeMillis() - lastKpiEventSent) > kpiFlushInterval) {
+            sendKpi();
+            lastKpiEventSent = System.currentTimeMillis();
+        }
+    }
+
+    public void recordBatchSent(final String eventType, final int bytesCount, final int eventsCount) {
+        final StreamKpiData kpiData = kpiDataPerEventType.computeIfAbsent(eventType, (x) -> new StreamKpiData());
+        kpiData.addBytesSent(bytesCount);
+        kpiData.addNumberOfEventsSent(eventsCount);
+        kpiData.incBatchesCount();
```

Comment:
Yeah, this could be a single method that takes two parameters and increments batch count in one go.